src/main.rs

```diff
@@ -792,51 +844,61 @@ fn bisect_nightlies(cfg: &Config, client: &Client) -> Result<BisectionResult, Er
         if t.is_current_nightly() {
             eprintln!("checking {} from the currently installed default nightly \
                        toolchain as the last failure", t);
-        } else {
-            eprintln!("checking {}", t);
         }
-        match t.install(client, &dl_spec) {
-            Ok(()) => {
-                eprintln!("verifying the start of the range does not reproduce the regression");
-                let outcome = t.test(&cfg);
-
-                if !cfg.args.preserve {
-                    let _ = t.remove(&dl_spec);
-                }
 
-                if let TestOutcome::Baseline = outcome {
-                    eprintln!("confirmed {} does not reproduce the regression", t);
+        match install_and_test(&t, &cfg, &client, &dl_spec) {
+            Ok(r) => {
+                // If Satisfies::No, then the regression was not identified in this nightly.
+                // Break out of the loop and use this as the start date for the
+                // bisection range
+                if let Satisfies::No = r {
                     first_success = Some(nightly_date);
                     break;
                 } else if has_start {
-                    bail!("the start of the range to test must not reproduce the regression");
+                    // If this date was explicitly defined on CL & has regression,
```

Comment:
I don't think the abbreviation "CL" is common enough to use it here. Please spell out "command line", assuming that *is* what this stood for.

Reply:
(you might win me over if you wrote "CLI" instead of  "CL"; but the best answer is to not use an acronym at all here.)

Reply:
Will do!

Reply:
No problem to write out "command line".  I will do that

Reply:
c950c79319382cb9beff5dacace48ca3ebc23d06